### PR TITLE
Lockhammer unittest and sweeptest

### PR DESCRIPTION
**Pull request by zoybai**
By default, test_lockhammer.py will run unittest specified by lh_test_cfg.yaml

If all tests have been passed, test_lockhammer.py will return 0. Otherwise any failed test will trigger a non-zero return value.

### CODE REVIEW

**Comment by geoffreyblake** [comment]
Can one of the admins verify this patch?

**Comment by rpgolshan** [comment]
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/Build_and_Test_PR-31-JOSCI5ULR755CP4TYEF6YZ65KRMJNW6DBDQA7TRM62VOIMSIHOBA/benchmarks/lockhammer/scripts/test_lockhammer.py", line 307, in <module>
    lhConfig = read_config(LH_CFG)
  File "/var/lib/jenkins/workspace/Build_and_Test_PR-31-JOSCI5ULR755CP4TYEF6YZ65KRMJNW6DBDQA7TRM62VOIMSIHOBA/benchmarks/lockhammer/scripts/test_lockhammer.py", line 60, in read_config
    with open(lhCfgStr, 'r') as fd:
FileNotFoundError: [Errno 2] No such file or directory: 'lh_test_cfg.yaml'

**Comment by zoybai** [comment]
jenkins issue has been fixed in the latest patch

File: benchmarks/lockhammer/scripts/lh_test_cfg.yaml [hunk 1]
```diff
@@ -34,8 +34,8 @@
 
 ## Global Settings
 globalcfg:
-    execdir: /tmp/synchronization-benchmarks/benchmarks/lockhammer/build
-    logfile: /tmp/lockhammer.csv
+    execdir: /root/synchronization-benchmarks/benchmarks/lockhammer/build
```
Comment (geoffreyblake) [inline benchmarks/lockhammer/scripts/lh_test_cfg.yaml]: You cannot hard-code the paths for the unit-tests.  These paths need to be automatically generated in the code to be relative to wherever the code is being run from, either in Jenkins or someone's workstation.

**Comment by zoybai** [comment]
Fixed path handling code:

Now by default the output lockhammer.csv file is a relative directory. It will be created in the working directory where lockhammer test script is executed.

The execute binary directory is also relative to where the unittest script is installed, just like config yaml file directory. We will prefix the absolute address of unittest python script __file__ to the binary relative directory ../build

**Comment by zoybai** [comment]
Passed on Jenkins test:
Stage Logs (Test)
 Shell Script -- git config --get remote.origin.url > .git/remote-url -- (self time 263ms)
Read file from workspace (self time 9ms)
 Set GitHub commit status (universal) (self time 174ms)
 Shell Script -- /var/lib/jenkins/workspace/Build_and_Test_PR-31-JOSCI5ULR755CP4TYEF6YZ65KRMJNW6DBDQA7TRM62VOIMSIHOBA/benchmarks/lockhammer/scripts/test_lockhammer.py -- (self time 1min 42s)
 Shell Script -- git config --get remote.origin.url > .git/remote-url -- (self time 264ms)
Read file from workspace (self time 14ms)
 Set GitHub commit status (universal) (self time 480ms)

**Pull request merged**